### PR TITLE
Clean up rustdoc tests by moving them into one place

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #76035 

The idea behind this move is to have tests in one place. It became very important for me because we started to have fixtures, making the code "more messy" than it should in my opinion. Like this, all tests are in one place and they're not mixed with the code anymore.

r? @jyn514 

I think it'll also interest @ollie27 so pinging them!

### CODE REVIEW

File: src/librustdoc/rustdoc_tests/mod.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+mod html_highlight;
+#[cfg(test)]
+mod html_markdown;
+#[cfg(test)]
+mod html_render;
+#[cfg(test)]
+mod html_toc;
+#[cfg(test)]
+mod passes_unindent_comments;
+#[cfg(test)]
```
Comment (matklad) [inline src/librustdoc/rustdoc_tests/mod.rs:15]: Should we just cfg(test) the parent module instead of duplicating the cfgs?

**Comment by petrochenkov** [comment]
The whole point of Rust unit tests is that they can be placed in modules which they are testing, preventing things that don't need to be public from being public.
No other rustc crate uses the proposed scheme.

**Comment by jyn514** [comment]
I agree with petrochenkov here ... this makes a _lot_ of functions public just for testing.

> It became very important for me because we started to have fixtures, making the code "more messy" than it should in my opinion.

Can you expand on this? What fixtures does rustdoc use for unit tests?

**Comment by jyn514** [comment]
Also, you linked to the wrong issue in the PR description, I think you meant https://github.com/rust-lang/rust/issues/76036.

**Comment by jyn514** [comment]
I'm going to close this since it's not clear it's a useful change.

**Pull request closed without merging**